### PR TITLE
[LIB-10618] Add d3-sankey.d.ts && support for flow chart

### PR DESCRIPTION
**Pull request by krzysztofKubiakVizlib**
Jira: [LIB-10618]
Jira: [LIB-10619]

[LIB-10618]: https://vizlib.atlassian.net/browse/LIB-10618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIB-10619]: https://vizlib.atlassian.net/browse/LIB-10619?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,8 @@
     "eslint": "6",
     "rollup": "1",
     "rollup-plugin-terser": "5",
-    "tape": "4"
+    "rollup-plugin-copy": "^3.4.0",
+    "tape": "4",
+    "d3": "*"
```
Comment (KamilPodsiadlo) [inline package.json]: d3 is redundant

File: src/sankey.js [hunk 2]
```diff
@@ -191,6 +197,39 @@ export default function Sankey(config) {
     }
   }
 
+  /**
+   * Compute nodes with fixed position
+   * @param nodes
+   * @returns {any[]}
+   */
```
Comment (KamilPodsiadlo) [inline src/sankey.js:203]: it's not necessary I think

File: src/sankey.js [hunk 3]
```diff
@@ -191,6 +197,39 @@ export default function Sankey(config) {
...
+  function computeFixedNodeLayer({nodes}) {
+    const d3group = group(nodes, d => d.dimIndex);
```
Comment (KamilPodsiadlo) [inline src/sankey.js]: rename please to something like `groupedNodes`

File: src/sankey.js [hunk 4]
```diff
@@ -191,6 +197,39 @@ export default function Sankey(config) {
...
+    const columnsFlow = [];
```
Comment (KamilPodsiadlo) [inline src/sankey.js]: As it's now separate method, it can be named just `columns`

File: src/sankey.js [hunk 5]
```diff
@@ -191,6 +197,39 @@ export default function Sankey(config) {
...
+
+    d3group.forEach(el => {
+      columnsFlow.push(el);
+    });
+
+    const x = config.fixedNodeLayerNumber;
+    const kx = (x1 - x0 - dx) / (x - 1);
+
+    const columns = [...columnsFlow];
```
Comment (KamilPodsiadlo) [inline src/sankey.js]: When you change `columnsFlow` into `columns`, then these line will be safe to delete I think

**Review by KamilPodsiadlo — CHANGES_REQUESTED** [review]
comments

File: src/sankey.js (hunk 2 above)
Comment (krzysztofKubiakVizlib) [reply src/sankey.js:203]: fixed

File: src/sankey.js (hunk 3 above)
Comment (krzysztofKubiakVizlib) [reply src/sankey.js]: changed

File: package.json (hunk 1 above)
Comment (krzysztofKubiakVizlib) [reply package.json]: removed

**Review by KamilPodsiadlo — APPROVED** [review]

File: src/sankey.js (hunk 4 above)
Comment (krzysztofKubiakVizlib) [reply src/sankey.js]: 👍

File: src/sankey.js (hunk 5 above)
Comment (krzysztofKubiakVizlib) [reply src/sankey.js]: 👍

**Pull request merged**